digitalocean/datasource_digitalocean_region.go

```diff
@@ -0,0 +1,72 @@
+package digitalocean
+
+import (
+	"fmt"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceDigitalOceanRegion() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDigitalOceanRegionRead,
+		Schema: map[string]*schema.Schema{
+			"slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sizes": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"features": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"available": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceDigitalOceanRegionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).godoClient()
+
+	regions, err := getDigitalOceanRegions(client)
+	if err != nil {
+		return fmt.Errorf("Unable to load regions: %s", err)
+	}
+
+	slug, ok := d.GetOk("slug")
+	if !ok || slug == "" {
```

Comment:
nit: is this necessary since `slug` is required? Could we use a validator here?

Reply:
We can. Pushing fix now.